test/commands/test_update_profile_from_orcid_record.py

```diff
@@ -52,6 +52,39 @@ def test_it_updates_the_name():
     assert profile.name.index == 'Family Name, Given Names'
 
 
+def test_it_does_not_updates_the_name_if_its_missing():
+    profile = Profile('12345678', Name('Old Name'))
+    orcid_record = {'person': {'name': {}}}
+
+    update_profile_from_orcid_record(profile, orcid_record)
+
+    assert profile.name == Name('Old Name')
+
+
+def test_it_updates_the_name_if_there_is_no_given_name():
```

Comment:
Through the ORCID UI it appears to be impossible to trigger this (a private name seems not to have the `name` section at all). _But_, the XSD seems to say that it is possible. 🤷‍♂️